src/engine/Operation.h

```diff
@@ -282,12 +283,17 @@ class Operation {
   // its result.
   [[nodiscard]] virtual bool supportsLimit() const { return false; }
 
-  // Hook subclasses can override to pass changes to the limit to their children
+ private:
+  // This function is called each time `applyLimit` is called. It can be
+  // overridden by subclasses to e.g. implement the LIMIT in a more efficient
+  // way
   virtual void onLimitChanged(const LimitOffsetClause&) const {}
 
+ public:
   // Set the value of the `LIMIT` clause that will be applied to the result of
-  // this operation. It will not replace any previous limit, stacking on top
-  // instead.
+  // this operation. If a LIMIT was previously set, this limit will not be
+  // replaced, but the new LIMIT will be applied additionally after the previous
```

Comment:
As those functions don't only refer to `LIMIT` but also to `OFFSET`, maybe now is the time to adjust their names + comments (`applyLimitOffset` and `onLimitOffsetChanged` to make the code more readable.